File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    internal int ExpectedMinutesInOven() => 40;

    internal int RemainingMinutesInOven(int timeInOven) => ExpectedMinutesInOven() - timeInOven;

    internal int PreparationTimeInMinutes(int layers) => layers * 2;

    internal int ElapsedTimeInMinutes(int layers, int timeInOven) => PreparationTimeInMinutes(layers) + timeInOven;
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40
let minutesPerLayer = 2

let remainingMinutesInOven (timeInOven: int): int =
    expectedMinutesInOven - timeInOven

let preparationTimeInMinutes (layers: int) : int =
    minutesPerLayer * layers

let elapsedTimeInMinutes (layers: int) (timeInOven: int): int =
    preparationTimeInMinutes(layers) + timeInOven
